### ISSUE TITLE
DenoisingAutoEncoder layer

### ISSUE THREAD

**Issue by Reddine** [fchollet/keras#336]
Hi,
I write this issue to notice a needed refactoring of DenoisingAutoEncoder layer and see if my implementation of denoising auto-encoder is right .
I am trying to implement Stacked Denoising Autoencoders with Keras but the DenoisingAutoEncoder cannot be the first layer in a pipeline as the documentation shows. That first layer make an additional weights which doesn't help in stacking DAE. 
I followed the mentioned example in the documentation of the DenoisingAutoEncoder and add a hello print command inside corrupt_input but the text never get printed.
To implement SDAE, I used AutoEncoder with a Masked layer(Dropout or/and GaussianNoise) as a first layer in the container encoder.
Also, I tried to fix DenoisingAutoEncoder (making a mess in the code architecture of Keras ^^) with this code in init and remove the _corrupt_input and get_input methods:
```python
if hasattr(encoder,'layers'):
    encoder.layers[:0] = [Dropout(corruption_level)]
    encoder.layers[1].set_previous(encoder.layers[0])
else:
    from ..layers import containers
    encoder = containers.Sequential([Dropout(corruption_level),encoder])
```
Cheers,

**Comment by fchollet** [comment]
I just removed the denoising autoencoder. Denoising is an application of the autoencoder, and I don't think we should be mixing applications and implementations. Besides, the implementation was incorrect (it was behaving the same as the regular autoencoder). This escaped me when I accepted the PR apparently.

**Comment by Imorton-zd** [comment]
Hi, @Reddine , have you  implemented the DenoisingAutoEncoder by keras? I want to use auto-encoder for texts classification, would you give me some suggestions?

**Comment by Reddine** [comment]
Hi, I switched to pylearn2 library for Denoising AutoEncoder purpose, It contains many built-in functions for classification and it s based on theano framework as keras. Notice that pylearn2 project does not have any current developer.
